File: android/src/main/java/expo/modules/screencorners/ExpoScreenCornersModuleSharedObject.kt
package expo.modules.screencorners

import expo.modules.kotlin.AppContext
import expo.modules.kotlin.sharedobjects.SharedObject

class ExpoScreenCornersModuleSharedObject(appContext: AppContext) : SharedObject(appContext) {
  var count: Int = 0

  override fun sharedObjectDidRelease() {
    super.sharedObjectDidRelease()
  }
}

File: src/ExpoScreenCornersModuleSharedObject.ts
import { SharedObject, useReleasingSharedObject } from 'expo-modules-core';

import ExpoScreenCornersModule from './ExpoScreenCornersModule';

export declare class ExpoScreenCornersModuleSharedObject extends SharedObject {
  count: number;
}

/**
 * Creates a new ExpoScreenCornersModuleSharedObject instance.
 * You are responsible for releasing it from memory by calling `release()` when done.
 */
export function createExpoScreenCornersModuleSharedObject(): ExpoScreenCornersModuleSharedObject {
  return new ExpoScreenCornersModule.ExpoScreenCornersModuleSharedObject();
}

/**
 * A hook that creates a ExpoScreenCornersModuleSharedObject instance and automatically
 * releases it when the component unmounts.
 */
export function useExpoScreenCornersModuleSharedObject(): ExpoScreenCornersModuleSharedObject {
  return useReleasingSharedObject(() => new ExpoScreenCornersModule.ExpoScreenCornersModuleSharedObject(), []);
}
